src/Serialization/Compact.php

```diff
@@ -77,13 +89,19 @@ protected function parsePayload($payloadJson)
 
     /**
      * @param string $jwt
+     *
      * @return Token
+     * @throws \InvalidArgumentException
      */
     public function deserialize($jwt)
     {
         $token = new Token();
 
-        list($encodedHeader, $encodedPayload, $encodedSignature) = explode('.', $jwt);
+        if (empty($jwt)) {
+            throw new \InvalidArgumentException('Not a valid JWT string passed for deserialization');
+        }
+
+        list($encodedHeader, $encodedPayload, $encodedSignature) = array_pad(explode('.', $jwt, 3), 3, null);
```

Comment:
But it's not empty (like empty string or array), it's just `null`, like not initialized. 

I added a test example with token without a signature.